lib/withOnyx.js

```diff
@@ -100,7 +119,17 @@ export default function (mapOnyxToState) {
                     return;
                 }
 
-                this.setState({...this.tempState, loading: false});
+                const stateUpdate = {...this.tempState, loading: false};
+
+                // Why are we setting the state here manually, instead of just using setState?
+                // Explanation: setState is a async operation, meaning it might execute on the next task,
+                // or to a later point in the microtask queue. That means, that there is a chance that
+                // setWithOnyxState is called again before the state is actually set. This causes issues
+                // with the above checks for the loading state.
+                // Note: This behaviour has been mainly observed on fabric.
```

Comment:
```suggestion
                // The state is set here manually, instead of using setState because setState is an async operation, meaning it might execute on the next task,
                // or at a later point in the microtask queue. That can lead to a race condition where
                // setWithOnyxState is called before the state is actually set. This results in unreliable behavior when checking the loading state and has been mainly observed on fabric.
```